rclcpp/include/rclcpp/subscription.hpp

```diff
@@ -104,31 +112,47 @@ class SubscriptionBase
   virtual void
   return_message(std::shared_ptr<void> & message) = 0;
 
+  /// Return the message borrowed in create_raw_message.
+  /** \param[in] message Shared pointer to the returned message. */
+  virtual void
+  return_raw_message(std::shared_ptr<rcl_message_raw_t> & message) = 0;
+
   virtual void
   handle_intra_process_message(
     rcl_interfaces::msg::IntraProcessMessage & ipm,
     const rmw_message_info_t & message_info) = 0;
 
+  rosidl_message_type_support_t
+  get_message_type_support_handle() const;
```

Comment:
Should this return a const reference instead? Not sure it matters, but worth considering.